parser/src/python.lalrpop

```diff
@@ -335,7 +353,7 @@ IfStatement: ast::Statement = {
 };
 
 WhileStatement: ast::Statement = {
-    <location:@L> "while" <test:Test> ":" <body:Suite> <s2:("else" ":" Suite)?> => {
+    <location:@L> "while" <test:NamedExpression_Test> ":" <body:Suite> <s2:("else" ":" Suite)?> => {
```

Comment:
Thanks for sticking with the grammar as described here: https://docs.python.org/3/reference/grammar.html?highlight=grammar

Reply:
When keeping Test and just extending it with named expressions, the grammar is too permisse and allows plentyof situations you need to find in a later analysis.  E.g., it is not allowed to use a named expression as expression statement, at least without braces. Thought, there  are other ways to achieve similarin the grammar, but for compatibility I would stick to that solution. 
For the the implementation, I consider [this grammar](https://github.com/python/cpython/blob/master/Grammar/python.gram) as more use helpful inspiration.
The naming sceme is violated.